packages/flutter_tools/test/general.shard/dart/sdk_validation_test.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/dart/analysis.dart';
+import 'package:flutter_tools/src/dart/pub.dart';
+import 'package:flutter_tools/src/dart/sdk.dart';
+import 'package:meta/meta.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+void main() {
+  group('sdk validation', () {
+    AnalysisServer server;
+    Directory tempDir;
+
+    setUpAll(() {
+      Cache.disableLocking();
+      tempDir =
+          fs.systemTempDirectory.createTempSync('sdk_validation_test').absolute;
```

Comment:
@devoncarew @zanderso 
nits: please see these sections of the style guide
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#have-good-hygiene-when-using-temporary-directories